pharapeti/group-arrangement
# Auth/Sessions/Routing

## What is this change doing?

- Fixes model associations
- Sets up routes, see `server.js` - i've split up the routes into /auth, /admin and /student
- Using encrypted password instead of plaintext for authentication via `bcryptjs`
- Persistent session store, allows a user to stay logged in after server restart

## Why is it needed?

In same order as above

- So we can use the associations to be able to make complex queries
- Allows us to break up the authentication, admin and student related endpoints - less complexity
- So we aren't storing plaintext passwords in the db
- Acts more like a real web app

To authenticate users + routing for future endpoints

## Where should the reviewer start?

server.js

## How can these changes be manually tested?

1) Use postman and check that (most 😉 ) endpoints are returning the right data
2) Ensure you stay logged in even after restarting the server
3) Ensure no plaintext passwords are being stored in the db after logging in

## What user story does this PR relate to? (include link)

Bunch of user stories
